controller/src/main/java/io/pravega/controller/store/stream/PravegaTablesStreamMetadataStore.java

```diff
@@ -285,6 +286,31 @@ PravegaTablesScope newScope(final String scopeName) {
                 executor);
     }
 
+    @Override
+    public CompletableFuture<Void> addStreamTagsToIndex(String scope, String streamName, StreamConfiguration config,
+                                                        OperationContext ctx, Executor executor) {
+        OperationContext context = getOperationContext(ctx);
+        if (streamName.startsWith(NameUtils.INTERNAL_NAME_PREFIX)) {
+            // Tags are not allowed on internal streams.
+            return CompletableFuture.completedFuture(null);
+        } else {
+            return Futures.completeOn(((PravegaTablesScope) getScope(scope, context))
+                                              .addTagsUnderScope(streamName, config.getTags(), context), executor);
+        }
+    }
+
+    @Override
+    public CompletableFuture<Void> removeTagsFromIndex(String scope, String streamName, Set<String> tagsRemoved,
+                                                       OperationContext ctx, Executor executor) {
+        OperationContext context = getOperationContext(ctx);
+        if (streamName.startsWith(NameUtils.INTERNAL_NAME_PREFIX)) {
+            // Tags are not allowed on internal streams.
+            return CompletableFuture.completedFuture(null);
+        } else {
+            return Futures.completeOn(((PravegaTablesScope) getScope(scope, context))
```

Comment:
what happens when the last stream with the tag is removed. do we also remove the tag entry?

Reply:
Good point, we are not removing the tag entry from the Table if the TagRecord has zero streams. Will improve this logic.

Reply:
done.